### ISSUE TITLE
Plugin not compatible with Gatsby V3

### ISSUE THREAD

**Issue by LarsEjaas** [d4rekanguok/gatsby-schema-field-absolute-path#10]
I am struggling to get this brilliant solution working. 

The plugin is throwing an error in the terminal when installing, and does not seem to like gatsby v.3 . I tried forcing an install, but am unsure if I am doing anything wrong but I keep getting null in the newly added image field.

I am unsure if I am doing something wrong regarding the filepath, or this is because of the new gatsby-image-plugin ? 

I have looked a lot at your example respository and blogpost: https://www.byderek.com/post/a-stackoverflow-question--a-use-case-for-gatsbys-field-extension but I still keep getting "null" in the newly created field.

**Comment by d4rekanguok** [comment]
Hi @LarsEjaas! Sorry to hear that the plugin hasn't worked for you.

I have just tried out the plugin on a simple Gatsby v3 project & it seems to work, but I understand that setting this up is confusing. I've added a verbose logging option, hopefully that'll help identifying the issue. Please see here:
https://github.com/d4rekanguok/gatsby-schema-field-absolute-path#enable-logging

I haven't tried out the new gatsby-image-plugin, but this plugin only touch the file paths so it shouldn't matter.

If you still have issues, would you be able to set up a minimal reproducable repo? I can take a look.

**Comment by LarsEjaas** [comment]
Hi again!

Thanks for the fast reply! I took another cup of coffee and added some more console.logs in my gatsby-node file to get an idea of what's going wrong here!

The things is: I got a lot of things going on in my gatsby-node file (for example my node file is checking markdown-field "published" for true or false to know if this should be included in the build)  and it looks like the "createSchemaCustomization" is actually run before the markdown article files are converted to pages AND even before the markdown file is converted with the template file - isn't that the reason for null values for all image fields would you think?.

Here's a copy of my nodefile for reference - I have both tried running the code with and without the plugin - but still get null values.

```
const config = require('./gatsby-config');

const { createFilePath } = require(`gatsby-source-filesystem`);

const path = require('path')

exports.onCreateNode = ({ node, getNode }) => {
  if (node.internal.type === `MarkdownRemark`) {
    const fileNode = getNode(node.parent)
    console.log('this is the relative path to the markdown article:',`\n`, fileNode.relativePath)
  } 
}

exports.createPages = async ({ actions, graphql, reporter }) => {
  const { createPage } = actions
  const articleTemplate = require.resolve(`./src/templates/articleTemplate.js`)
  const result = await graphql(`
    {
      allMarkdownRemark(
        sort: { order: DESC, fields: [frontmatter___date] }
        limit: 1000
      ) {
        edges {
          node {
            frontmatter {
              slug
              published
            }
          }
        }
      }
    }
  `)
  // Handle errors
  if (result.errors) {
    reporter.panicOnBuild(`Error while running GraphQL query.`)
    return
  }
  const allPosts = result.data.allMarkdownRemark.edges;

  const allowedPosts = allPosts.filter(post => 
    (process.env.NODE_ENV === 'development' 
    || post.node.frontmatter.published));

  allowedPosts.forEach(({node}) => {
    console.log('page now created from article template for:', node.frontmatter.slug);
    createPage({
      path: node.frontmatter.slug,
      component: articleTemplate,
      context: {
               // additional data can be passed via context
               slug: node.frontmatter.slug,
      },
    })
  })
}


exports.createSchemaCustomization = ({ actions }) => {
  const { createFieldExtension, createTypes } = actions

  createFieldExtension({
    name: 'fileByDataPath',
    extend: () => ({
      resolve: function (src, args, context, info) {
        const partialPath = src.cover_image
          if (!partialPath) {
            return null
          }

        const filePath = path.join(__dirname, 'src/articles/images/', partialPath)
        console.log("This is the filepath used to source image: ", filePath);
        const fileNode = context.nodeModel.runQuery({
          firstOnly: true,
          type: 'File',
          query: {
            filter: {
              absolutePath: {
                eq: filePath
              }
            }
          }
        })

        if (!fileNode) {
          return null
        }
        console.log('this is the partial path:', partialPath);
        return fileNode
      }
    })
  })

  const typeDefs = `
    type Frontmatter @infer {
      coverImage: File @fileByImagesPath
    }

    type MarkdownRemark implements Node @infer {
      frontmatter: Frontmatter
    }
  `

  createTypes(typeDefs)
}

//coverImage: File @fileByDataPath


```

**Comment by LarsEjaas** [comment]
Hi again Derek. 
I ended up going another route with this. 

I made an image component that sources the images from the folder I needed them in (my GraphQL sorting files by  sourceInstanceName - as a reference to the name I gave the source-fileSystem in gatsby-config) and sourced the image from this list. It's probably not scalable to large sites, but works well for my use. 

I used some time to debug why I couldn't attach images to the articles, but couldn't find the root cause unfortunately.

Thanks for taking the time to help!
